### DISCUSSION TITLE
Enhancements - Automatic backfill of second/minute data, customization of the "detailed" tag name and values via the config file

### DISCUSSION THREAD

**Discussion by cooldil** [jertel/vuegraf#172] [Ideas]
Have added some enhancements (and fixes) to the code; namely:

FIXES:
- Time precision on datapoints submitted to the database for the second, minute and hour dataseries.  Prevents duplication of points if the data is pulled and stored again in the future.
- Addresses the UTC offset on the datapoint timestamp for the Day series during historical pulls.
- Saves historical Day/Hour datapoints to the databases as each batch is pulled from Emporia.  Prevents timeout issues while submitting to the database if a large window is provided for the historydays parameter. 

ENHANCEMENTS:
- Allows for the customization of the "detailed" tag name and values via the tagName, tagValue_second, tagValue_minute, tagValue_hour and tagValue_day options in the config file.  Changes should be fully backwards compatible.
- Automatically backfills the second/minute data if the database is missing any datapoints.  Second data can be backfilled for a maximum of 3 hours while the minute data can be backfilled for a maximum of 7 days.
- Allows for the customization of the maximum number of history days that can be pulled via the maxHistoryDays config file entry.

The changes can be reviewed at [https://github.com/jertel/vuegraf/compare/master...cooldil:vuegraf:cooldil-patch-2?diff=split&w=](url) and then when/if appropriate I can submit as a pull request.

**Comment by jertel** [comment]
Thanks for sharing your changes! Particularly the ability to adjust the `detailed` field name and values. That field started out as a simple bool flag and later community contributions added to it so it's become a bit inconsistent, as you noticed. I'd like to incorporate your changes into the project. Please submit a PR. The README.md will need updated to reflect the enhancement usage instructions. 

This has grown from a little script into a larger app so someday soon we should properly refactor and split it into smaller files with unit tests.

**Reply by cooldil** [reply]
PR submitted.  
Let me know if you have any questions regarding any of the code and/or if any changes need to be made.  Did allow you via the PR to make changes in the branch if you so desire before merging.

Point taken regarding the complexity of the code as it's grown.  I was having a tough time keeping my own customizations within my fork consistent with changes you were making :-)
